Recurly/JsonSerializer.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using Newtonsoft.Json.Converters;
+using RestSharp.Serializers;
+using RestSharp.Deserializers;
+using RestSharp;
+using System.IO;
+
+namespace Recurly {
+    public class JsonSerializer : ISerializer, IDeserializer
+    {
+        private Newtonsoft.Json.JsonSerializer serializer;
+        private List<string> _errors;
+        public List<string> Errors {
+            get {
+                if (this._errors == null)
+                    this._errors = new List<string>();
+                return this._errors;
+            }
+        }
+        
+        public string RootElement { get; set; }
+        public string Namespace { get; set; }
+        public string DateFormat { get; set; }
+        public string ContentType { get; set; }
+
+        public JsonSerializer() {
+            ContentType = "application/json";
+            var contractResolver = new DefaultContractResolver
+            {
+                NamingStrategy = new SnakeCaseNamingStrategy()
+            };
+            var settings = new JsonSerializerSettings() {
+                ContractResolver = contractResolver,
+                NullValueHandling = NullValueHandling.Ignore,
+                MissingMemberHandling = MissingMemberHandling.Ignore,
+                DateFormatHandling = DateFormatHandling.IsoDateFormat,
+                DateTimeZoneHandling = DateTimeZoneHandling.Utc,
+                Error = delegate(object sender, Newtonsoft.Json.Serialization.ErrorEventArgs args)
+                {
+                    Errors.Add(args.ErrorContext.Error.Message);
+                    args.ErrorContext.Handled = true;
+                },
+                //Formatting = Formatting.Indented,
```

Comment:
Same here, do we want to leave this in?

Reply:
Until we have better logging, this one is helpful to have on hand.